src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/ResourceUpdaterTests.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+using ILCompiler.Win32Resources;
+using Microsoft.NET.HostModel;
+using Xunit;
+
+namespace AppHost.Bundle.Tests;
+
+public class ResourceUpdaterTests
+{
+    private MemoryStream CreateTestPEFileWithoutRsrc()
+    {
+        var peBuilder = new ManagedPEBuilder(
+            PEHeaderBuilder.CreateExecutableHeader(),
+            new MetadataRootBuilder(new MetadataBuilder()),
+            ilStream: new BlobBuilder());
+        var peImageBuilder = new BlobBuilder();
+        peBuilder.Serialize(peImageBuilder);
+        var memoryStream = new MemoryStream();
+        memoryStream.Write(peImageBuilder.ToArray());
+        memoryStream.Seek(0, SeekOrigin.Begin);
+
+        memoryStream.Seek(0, SeekOrigin.Begin);
+
+        return memoryStream;
+    }
+
+    private MemoryStream GetCurrentAssemblyMemoryStream()
+    {
+        var memoryStream = new MemoryStream();
+        memoryStream.Write(File.ReadAllBytes(Assembly.GetExecutingAssembly().Location));
+        memoryStream.Seek(0, SeekOrigin.Begin);
+
+        memoryStream.Seek(0, SeekOrigin.Begin);
+
+        return memoryStream;
+    }
+
+    [Fact]
+    void ResourceUpdaterAddResourceToPEWithoutRsrc()
+    {
+        using var memoryStream = CreateTestPEFileWithoutRsrc();
+
+        using (var updater = new ResourceUpdater(memoryStream, true))
+        {
+            updater.AddResource("Test Resource"u8.ToArray(), "testType", 0);
```

Comment:
Can you also add cases for:
- Using a ushort as the type instead of a string
- Adding multiple resources with the same type, different name
- Copying resources via `AddResourcesFromPEImage`